### ISSUE TITLE
2.3.1: Existing booking page crashes when set active

### ISSUE THREAD

**Issue by Sarmingsteiner** [SergeyMosin/Appointments#564]
After updating to 2.3.1, our only booking page is crahing (NC shoes internal server error messge).
I found out that preview is working when the booking page is set inactive via settings page. On activating, it crashes again.

I tried a downgrade to 2.2.0, but the settings page is not loading at all and the booking page visible to site visitors shows a strange style and is not working.

Creating a new booking page works.
Any ideas (the configuration of a new booking page is a lot of work since we have many customisations on it)? 🙏

**Comment by SergeyMosin** [comment]
can you check NC logs for any errors

**Comment by SergeyMosin** [comment]
> a certain calendar in
Is there anything special about this calendar ?
Is it shared ? I could try to replicate but I need more information or a related error for nextcloud logs (if there is one)

**Comment by bentwnghk** [comment]
I am experiencing the same issue. My Nextcloud logs show:

```
[PHP] 錯誤: Attempt to read property "DTSTART" on null at /var/www/nextcloud/apps/appointments/lib/Backend/BCSabreImpl.php#447
	GET /apps/appointments/form?p=p0&v=1732405743574
[index] 錯誤: Call to a member function hasTime() on null in file '/var/www/nextcloud/apps/appointments/lib/Backend/BCSabreImpl.php' line 447
	GET /apps/appointments/form?p=p0&v=1732405743574
```

**Comment by Sarmingsteiner** [comment]
> > a certain calendar in
> 
> Is there anything special about this calendar ? Is it shared ? I could try to replicate but I need more information or a related error for nextcloud logs (if there is one)

Indeed, the non-working calendar is a shared calendar. Furthermore, it cannot be used as a calendar for conflicts check anymore.

Here is the log:
`{"message":"Call to a member function getDateTime() on null in file '.../apps/appointments/lib/Backend/FakeIterator.php' line 22","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36 OPR/114.0.0.0","version":"30.0.2.2","exception":{"Exception":"Exception","Message":"Call to a member function getDateTime() on null in file '/.../apps/appointments/lib/Backend/FakeIterator.php' line 22","Code":0,"Trace":[{"file":".../lib/private/AppFramework/App.php","line":161,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Appointments\\Controller\\PageController"},"formbase"]},{"file":".../lib/private/Route/Router.php","line":302,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\Appointments\\Controller\\PageController","formbase",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"_route":"appointments.page.formbase"}]},{"file":".../lib/base.php","line":1001,"function":"match","class":"OC\\Route\\Router","type":"->","args":["/apps/appointments/form"]},{"file":".../index.php","line":24,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":".../lib/private/AppFramework/Http/Dispatcher.php","Line":146,"Previous":{"Exception":"Error","Message":"Call to a member function getDateTime() on null","Code":0,"Trace":[{"file":"/.../apps/appointments/lib/Backend/BCSabreImpl.php","line":463,"function":"__construct","class":"OCA\\Appointments\\Backend\\FakeIterator","type":"->","args":[null,{"__class__":"DateTimeZone"}]},{"file":"/.../apps/appointments/lib/Backend/BCSabreImpl.php","line":351,"function":"buildBusyTree","class":"OCA\\Appointments\\Backend\\BCSabreImpl","type":"->","args":[["27","14","13"],{"start":1732535096,"end":1735664400},{"__class__":"DateTime"},1732621496,1735599600,false]},{"file":"/.../apps/appointments/lib/Controller/PageController.php","line":1168,"function":"queryTemplate","class":"OCA\\Appointments\\Backend\\BCSabreImpl","type":"->","args":[{"enabled":true,"label":"entries"},{"__class__":"DateTime"},{"__class__":"DateTime"},"user","p0"]},{"file":"/.../apps/appointments/lib/Controller/PageController.php","line":687,"function":"showForm","class":"OCA\\Appointments\\Controller\\PageController","type":"->","args":["base","user","p0"]},{"file":"/.../lib/private/AppFramework/Http/Dispatcher.php","line":208,"function":"formBase","class":"OCA\\Appointments\\Controller\\PageController","type":"->","args":[]},{"file":"/.../lib/private/AppFramework/Http/Dispatcher.php","line":114,"function":"executeController","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Appointments\\Controller\\PageController"},"formbase"]},{"file":"/.../lib/private/AppFramework/App.php","line":161,"function":"dispatch","class":"OC\\AppFramework\\Http\\Dispatcher","type":"->","args":[{"__class__":"OCA\\Appointments\\Controller\\PageController"},"formbase"]},{"file":"/.../lib/private/Route/Router.php","line":302,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OCA\\Appointments\\Controller\\PageController","formbase",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"_route":"appointments.page.formbase"}]},{"file":"/.../lib/base.php","line":1001,"function":"match","class":"OC\\Route\\Router","type":"->","args":["/apps/appointments/form"]},{"file":"/.../index.php","line":24,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"/.../apps/appointments/lib/Backend/FakeIterator.php","Line":22},"message":"Call to a member function getDateTime() on null in file '/.../apps/appointments/lib/Backend/FakeIterator.php' line 22","exception":{},"CustomMessage":"Call to a member function getDateTime() on null in file '/.../apps/appointments/lib/Backend/FakeIterator.php' line 22"}}
`

**Comment by SergeyMosin** [comment]
Fixed in v2.3.2
